### DISCUSSION TITLE
개발자님 항상 감사합니다! 질문이 있습니다

### DISCUSSION THREAD

**Discussion by rladbssam317** [changdoc/naver-vaccine-macro#236] [Q&A]
안녕하세요 개발자님! 좋은 파일 만들어 주셔서 감사합니다! 
네이버맵은 10분 정도 사용하면 graphql이 빨갛게 변해서 사용을 못하고 카카오맵을 사용하고 있습니다! 

질문이 두가지 있는데요 

1. F12 network에 left count도 정상적으로 fetch되고 있습니다. 그러나 분명 핸드폰으로는 잔여백신 알림 배너가 왔는데도 불구하고 카카오맵에서는 인식하지 못하였습니다. 왜 그러는걸까요? 인식되는 매커니즘이 어떻게 되는건지 궁금합니다. 지도상에 마감이라고 되어있으면 인식을 못하는건지, 아니면 단순 렉인건지 궁금합니다. 

2. 현재 4일동안 하고 있는데 어제 딱 한번 성공을 했어요. 근데 저희 부모님이 이미 아스트라로 예약이 되어있어서 중복예약 때문에 아쉽게 넘어갔습니다. 그래서 아스트라 우선 취소했어요. 우선 매크로가 잘 작동되고 있긴 한데 번번히 실패하고 있습니다. 혹시 여기서 예약 속도를 더 높일 수 있는 방법은 없나요? 저번에 어떤분이 500ms에서 250ms로 바꿧다고 했는데 시도해보았으나 똑같았습니다...

**Comment by rladbssam317** [comment]
그리고 질문 하나 더 있는데 

혹시 매크로를 긴 시간 틀어놓으면 렉이 걸리는 경우가 많나요? 

만약 예약페이지로 넘어갔을 때 실패했다면 다시 네이버맵에서도 graphql response 추출하고 

카카오맵에서도 left count 추출하고 이 작업을 다 다시 해야되는건가요?

**Reply by changdoc** [reply]
1. fetch 내에서 left count 필드가 0이상이면 해당 병원의 orgCd로 저장해두신 graphql response에서 추출한 병원 목록에서 병원을 찾아 해당 병원의 예약 신청 url로 이동하는 형태로 동작합니다. console창을 켜두시면 목록에 없는 병원이 디텍팅 되는 경우 관련 로그를 출력하도록 해뒀습니다.
2. 새로고침 주기는 예약 속도와는 무관할 것으로 예상되네요. 예약 신청이 이루어질때 잔여 백신이 탐지되어 예약 신청 페이지로 접근한뒤 예약 신청 처리까지 진행이 되는데 여기서 다른 매크로들과 속도차이가 나면 실패를 하게 되는 상황 같네요
3. 장시간 틀어놓으면 성능에 영향을 주는지 따로 테스트는 안해봤는데요 시스템 리소스로 브라우저의 cpu, 메모리 사용량 측정을 한번 해보시면 장시간 동작시 변동이 있는지 없는지 파악 가능하실듯하네요.
4. 추출하신 병원목록 graphql response와 카카오맵 left count request body는 저장되어야 매크로가 동작을 하기 때문에 페이지를 벗어났다가 다시 접근해도 동일한 데이터가 유지되도록 되어 있습니다.(저장이 안되어 있으면 동작을 안 함)

**Reply by rladbssam317** [reply]
답변감사드립니다! 

1. graphql에서 지도 상 19개의 병원은 모두 추출이 된 거 확인하였는데요.  카카오맵으로 돌리는 과정에서 핸드폰에서는 알람이 울렸지만 매크로에서는 잡지 못하였어요. 이건 왜그럴까요?ㅠㅠ

**Reply by rladbssam317** [reply]
아 그리고! 질문이 하나 더 있습니다. 제가 검색을 모더나로 켜놓았는데 화이자 잔여가 나와있을 때 넘어간 적이 있어요. 그건 단순 오류인가요?

**Reply by changdoc** [reply]
1. api 결과값이 left count에는 잔여 백신이 존재하지 않는 상황인데 어떤 상황인지 예측이 안되네요. 조회 타이밍에 알림을 발생시킨 잔여 백신이 이미 예약 처리가 되었다면 발생가능하겠네요.

현재 맵에서 잔여 백신 탐지시에 백신 종류는 별도로 구분 하지 않고 있습니다. 예약 신청 단계에서만 선택 백신으로 시도하게 되어 있는데요.
-> 카카오맵 쪽 데이터는 구분 가능한것으로 파악중인데 수정 가능 할때 예약 시도 대상 백신이 아닐경우 무시하도록 하는 부분 추가해보도록 하겠습니다.

**Reply by rladbssam317** [reply]
항상 감사합니다!
